src/BuiltinHelper.js

```diff
@@ -83,23 +70,42 @@ class BuiltinHelper extends Helper {
         }
     }
 
+    get (assetId) {
+        let asset = null;
+        if (this.assets.hasOwnProperty(assetId)) {
+            /** @type{BuiltinAssetRecord} */
+            const assetRecord = this.assets[assetId];
+            asset = new Asset(assetRecord.type, assetRecord.id, assetRecord.format, assetRecord.data);
+        }
+        return asset;
+    }
+
+    cache (assetType, dataFormat, data, id) {
+        if (!dataFormat) throw new Error('Data cached without specifying its format');
+        if (id) {
+            if (this.assets.hasOwnProperty(id)) return id;
```

Comment:
This isn't safe for project files, where the same ID might change content over time. If it's helpful we could add a flag somewhere (on `AssetType`, for example) to indicate, per asset type, whether the asset ID is determined by its content or not.